internal/bminventory/inventory.go

```diff
@@ -464,9 +464,13 @@ func (b *bareMetalInventory) RegisterClusterInternal(
 	log := logutil.FromContext(ctx, b.log).WithField(ctxparams.ClusterId, id)
 	log.Infof("Register cluster: %s with id %s and params %+v", swag.StringValue(params.NewClusterParams.Name), id, params.NewClusterParams)
 	success := false
+	fullCluster := &common.Cluster{}
 	var err error
 	defer func() {
 		if success {
+			if b.stream != nil && err == nil {
+				_ = b.stream.Notify(ctx, fullCluster.Cluster)
```

Comment:
maybe only log the error as a warning so we keep track in case ``Notify`` errors

Reply:
In this specific implementation we are sending the notification asynchronously. Do you think it's worth adding the message anyway for future-proofing? My concern is that when reading the code we would consider this not failing because the message doesn't show, but really it could be failing with another (internal) message

Reply:
Even notify does *some action* synchronously (in addition to the asynchronous actions), so as long as it's returning an error we should show it in some way 
If the asynchronous actions have their respecting logs that's totally fine and expected. When reading the logs or the code it's expected we'll get into the thinking of asynchronous execution, if I understand you correctly